### PR TITLE
Use solana-cli config keypair in solana-keygen

### PR DESCRIPTION
**Pull request by CriesofCarrots**
#### Problem
It would be handy to use the keypair override from `solana get` in solana-keygen, especially as we move toward supporting hardware wallets in these tools. Desired flow something like:

```
$ solana set --keypair <keypair_filepath_or_path_to_hardware_wallet>
$ solana-keygen pubkey
E13eqyhs8uJvXtKfmTkt82tiEk1hbz3R2E7GgbMtEYBH
```

#### Summary of Changes
- ~~Import solana-cli::config into solana-keygen~~ Move solana-cli::config into its own crate for more general use
- Add `--config` arg to solana-keygen to specify config file location (default: ~/.config/solana/cli/config.yml)
- If keypair_path is set in existing config, use that keypair for `solana-keygen pubkey` and `solana-keygen verify`. (Swapped arg order in `verify` to make this work)
- Also, I made the `verify --pubkey` arg required, as it was tacitly so, and would panic if not provided.

(Addition)
I think that keypair arg handling, like `get_keypair_from_matches()` should probably be moved somewhere general: either into the new solana-cli-config crate or solana-clap-utils::keypair. I will do that in a separate PR as part of implementation in cli and/or solana-validator.

### CODE REVIEW

File: keygen/src/keygen.rs [hunk 1]
```diff
@@ -193,22 +199,37 @@ fn main() -> Result<(), Box<dyn error::Error>> {
...
-                        .value_name("PATH")
+                        .value_name("BASE58_PUBKEY")
                         .takes_value(true)
-                        .help("Path to keypair file"),
+                        .required(true)
+                        .help("Public key"),
                 )
                 .arg(
-                    Arg::with_name("pubkey")
+                    Arg::with_name("infile")
```
Comment (garious) [inline keygen/src/keygen.rs]: Not `--keypair` to mirror `solana address --keypair`?

**Comment by mvines** [comment]
oh next up: `solana-cli-config` crate so it can be used by `solana-validator` too :)

**Comment by CriesofCarrots** [comment]
> oh next up: `solana-cli-config` crate so it can be used by `solana-validator` too :)

@mvines, yep, already doing that as we type :)

**Comment by CriesofCarrots** [comment]
@mvines , post-merge comments welcome

**Pull request merged**